examples/notebooks/transformers_tagger.ipynb

```diff
@@ -0,0 +1,454 @@
+    "    attention_mask: torch.Tensor\n",
+    "    input_len: List[int]\n",
+    "    overflowing_tokens: Optional[torch.Tensor] = None\n",
+    "    num_truncated_tokens: Optional[torch.Tensor] = None\n",
+    "    special_tokens_mask: Optional[torch.Tensor] = None"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The wrapped tokenizer will take a list of lists as input (the texts) and will output a `TokensPlus` object containing the fully padded batch of tokens. The wrapped transformer will take a list of `TokensPlus` objects and will output a list of 2-dimensional arrays.\n",
+    "\n",
+    "1. **TransformersTokenizer**: `List[List[str]]` → `TokensPlus`\n",
+    "2. **Transformer**: `TokensPlus` → `List[Array2d]`\n",
+    "\n",
+    "> 💡 Since we're adding type hints everywhere (and Thinc is fully typed, too), you can run your code through [`mypy`](https://mypy.readthedocs.io/en/stable/) to find type errors and inconsistencies. If you're using an editor like Visual Studio Code, you can enable `mypy` linting and type errors will be highlighted in real time as you write code.\n",
+    "\n",
+    "To wrap the tokenizer, we register a new function that returns a Thinc `Model`. The function takes the name of the pretrained weights (e.g. `\"bert-base-multilingual-cased\"`) as an argument that can later be provided via the config. After loading the `AutoTokenizer`, we can stash it in the attributes. This lets us access it at any point later on via `model.get_attr(\"tokenizer\")`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import thinc\n",
+    "from thinc.api import Model\n",
+    "from transformers import AutoTokenizer\n",
+    "\n",
+    "@thinc.registry.layers(\"transformers_tokenizer.v0\")\n",
+    "def TransformersTokenizer(name: str) -> Model[List[List[str]], TokensPlus]:\n",
+    "    return Model(\n",
+    "        \"tokenizer\",\n",
+    "        _tokenizer_forward,\n",
+    "        attrs={\"tokenizer\": AutoTokenizer.from_pretrained(name)},\n",
+    "    )\n",
+    "\n",
+    "def _tokenizer_forward(model, texts: List[List[str]], is_train: bool):\n",
+    "    tokenizer = model.get_attr(\"tokenizer\")\n",
+    "    token_data = tokenizer.batch_encode_plus(\n",
+    "        [(text, None) for text in texts],\n",
+    "        add_special_tokens=True,\n",
+    "        return_token_type_ids=True,\n",
+    "        return_attention_masks=True,\n",
+    "        return_input_lengths=True,\n",
+    "        return_tensors=\"pt\",\n",
+    "    )\n",
+    "    return TokensPlus(**token_data), lambda d_tokens: d_tokens"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The forward pass takes the model and a list of lists of strings and outputs the `TokensPlus` dataclass and a callback to use during the backwards (which does nothing in this case)."
```

Comment:
duplicate "list of list of"

Reply:
I think it is actually a list of lists, i.e. `List[List[str]`, but it's kinda awkward to get across 🤔

Reply:
Maybe `list-of-lists of strings`?